python/lsst/ts/observatory/control/auxtel/atcalsys.py

```diff
@@ -0,0 +1,417 @@
+        power of self.whitelight_power W
+        """
+        self.rem.atwhitelight.evt_lampState.flush()
+
+        await self.rem.atwhitelight.cmd_turnLampOn.set_start(
+            power=self.whitelight_startup_power, timeout=self.long_long_timeout * 8
+        )
+
+    async def wait_for_chiller_temp_within_tolerance(self) -> None:
+        """Checks if chiller reaches set self.chiller_temperature within
+        set self.chiller_temp_tolerance_relative in
+        less than self.timeout_chiller_cool_down s.
+
+        Raises
+        ------
+        TimeOutError:
+            If the chiller doesn't reach self.chiller_temperature
+            within tolerance in self.timeout_chiller_cool_down s.
+
+        Note
+        ----
+        Taken from power_on_atcalsys.py
+        PAF: I think this can be improved.
+        """
+        start_chill_time = time.time()
+        while time.time() - start_chill_time < self.long_long_timeout * 8:
+            chiller_temps = await self.rem.atwhitelight.tel_chillerTemperatures.next(
+                flush=True, timeout=self.long_timeout
+            )
+            tel_chiller_temp = chiller_temps.supplyTemperature
+            self.log.debug(
+                f"Chiller supply temperature: {tel_chiller_temp:0.1f} deg "
+                f"[set:{chiller_temps.setTemperature} deg]."
+            )
+            if (
+                abs(chiller_temps.setTemperature - tel_chiller_temp)
+                / chiller_temps.setTemperature
+                <= self.chiller_temp_tolerance_relative
+            ):
+                chill_time = time.time() - start_chill_time
+                self.log.info(
+                    f"Chiller reached target temperature, {tel_chiller_temp:0.1f} deg "
+                    f"within tolerance, in {chill_time:0.1f} s."
+                )
+                break
+        else:
+            raise TimeoutError(
+                f"Timeout waiting after {self.long_long_timeout * 8} s "
+                f"for the chiller to chill to {chiller_temps.setTemperature} deg. "
+                f"Stayed at {tel_chiller_temp:0.1f} deg."
+            )
+
+    async def wait_for_lamp_to_warm_up(self) -> None:
+        """Confirms the white light source has warmed up and is on.
+
+        Raises:
+        ------
+        TimeOutError:
+            If the lamp fails to turn on after self.timeout_lamp_warm_up
+
+        Note
+        ----
+        Taken from power_on_atcalsys.py
+        PAF: I think this can be improved.
+        """
+        lamp_state = await self.rem.atwhitelight.evt_lampState.aget(
+            timeout=self.long_long_timeout
+        )
+        self.log.info(
+            f"Lamp state: {self.rem.atwhitelight.LampBasicState(lamp_state.basicState)!r}."
+        )
+        while lamp_state.basicState != self.rem.atwhitelight.LampBasicState.ON:
+            try:
+                lamp_state = await self.rem.atwhitelight.evt_lampState.next(
```

Comment:
You probably need to flush this event before getting into the while loop.

Reply:
added a flush a couple lines above.